src/graph/GoExecutor.cpp

```diff
@@ -602,66 +596,62 @@ void GoExecutor::onStepOutResponse(RpcResponse &&rpcResp) {
 }
 
 
-void GoExecutor::maybeFinishExecution(RpcResponse &&rpcResp) {
+void GoExecutor::maybeFinishExecution() {
     auto requireDstProps = expCtx_->hasDstTagProp();
 
     // Non-reversely traversal, no properties required on destination nodes
     // Or, Reversely traversal but no properties on edge and destination nodes required.
     // Note that the `dest` which used in reversely traversal means the `src` in foword edge.
     if (!requireDstProps) {
-        finishExecution(std::move(rpcResp));
-        return;
-    }
-
-    auto dstIdStatus = getDstIdsFromResp(rpcResp);
-
-    if (!dstIdStatus.ok()) {
-        doError(std::move(dstIdStatus).status());
+        finishExecution();
         return;
     }
 
-    auto dstids = std::move(dstIdStatus).value();
+    auto dstIds = getDstIdsFromResps(records_.begin() + recordFrom_ - 1, records_.end());
 
     // Reaching the dead end
-    if (dstids.empty()) {
+    if (dstIds.empty()) {
         onEmptyInputs();
         return;
     }
 
     DCHECK(requireDstProps);
     // Only properties on destination nodes required
-    fetchVertexProps(std::move(dstids), std::move(rpcResp));
+    fetchVertexProps(std::move(dstIds));
     return;
 }
 
 void GoExecutor::onVertexProps(RpcResponse &&rpcResp) {
     UNUSED(rpcResp);
 }
 
-StatusOr<std::vector<VertexID>> GoExecutor::getDstIdsFromResp(RpcResponse &rpcResp) const {
+std::vector<VertexID> GoExecutor::getDstIdsFromResps(std::vector<RpcResponse>::iterator begin,
```

Comment:
How about `std::vector<VertexID> GoExecutor::getDstIdsFromResps(uint32_t beginPos)` the beginPos is the first element of recordFrom.

Reply:
I think iterator is better.